versions/development/SPEC.md

```diff
@@ -1218,81 +1270,314 @@ $runtime = 'runtime' $ws* '{' ($ws* $runtime_kv $ws*)* '}'
+
+It must also be clarified that the purpose of this section is **NOT** to provide an execution engine a means to encode all sorts of required values that it needs in order to execute a task without setting any default values. This practice should strictly be considered an anti-pattern and should be avoided by execution engines
+
+The values within the `hints` section are arbitrary K/V pairs, and it is generally up to the execution engine to define what values they are interested in. The provided values can be simple (`String`,`boolean`,`number`...)  or more complex (`Arrays`,`hints object`)
+
+```wdl
+task foo {
+  ...
+  hints {
+		
+  }
+}
+```
+
+#### Reserved Keys
+
+Although `hints` are arbitrary, there are a number of keys which are reserved by the language in or oder to try and encourage interoperability of tasks and workflows between different execution engines. The list of Reserved keys is likely to grow, and in order to prevent name space collisions, engines should follow the [best practices](#conventions-and-best-practices) for specifying hints. 
+
+* **maxCpu**: Specify the maximum CPU to be provisioned for a task. It is up to the engine, whether or not it enforces the `maxCpu` hint. The type of this hint should be the same as `runtime.cpu`
+* **maxMemory**: Specify the maximum memory provisioned for a task. It is up to the engine, whether or not it enforces the `maxMemory` hint. the type of this hint should be the same as `runtime.memory`
+* **shortTask**: Tell the execution engine that this task is not expected to take long to execute, and therefore the engine can attempt to optimize the execution in whatever way it defines. This flag can be used to tell an engine to use the cost optimized instance types that many clouds provide (but are available only for a limited time). An example of this would be `preemptible` instances on `gcp` and `spot` instances on `aws`.
+* **localizationOptional**: Tell the execution engine that whenever possible it does not need to localize the defined `File` type inputs for this task. Important to note, is this directive should not have any impact on the success or failure of a task (ie it should run with or without localization). The type of this hint is a boolean value
```

Comment:
Note to anyone else who almost makes the same mistake as me:

Athough this global `localizationOptional` is like a less useful version of the Cromwell `parameter_meta` hint, keep reading because there's a separate `inputs` hints section which covers the specific-per-input version of this hint.